src/sidebar/services/test/groups-test.js

```diff
@@ -391,6 +426,102 @@ describe('groups', function() {
       });
     });
 
+    it('catches error when fetching the direct-linked annotation', () => {
+      const svc = service();
+
+      fakeSettings.annotations = 'ann-id';
+
+      fakeApi.profile.groups.read.returns(Promise.resolve([]));
+      fakeApi.groups.list.returns(
+        Promise.resolve([
+          { name: 'BioPub', id: 'biopub' },
+          { name: 'Public', id: '__world__' },
+        ])
+      );
+      fakeApi.annotation.get.returns(
+        Promise.reject(
+          new Error(
+            "404 Not Found: Either the resource you requested doesn't exist, \
+          or you are not currently authorized to see it."
+          )
+        )
+      );
+
+      // The user is logged out.
+      fakeAuth.tokenGetter.returns(null);
```

Comment:
I think I was trying to re-create the "this is a private group and you don't have permission when you are logged out" scenario but in retrospect, no it doesn't matter and it can fail for other reasons as well. I've simplified the test.